nodejs/eks/examples/storage-classes/utils.ts

```diff
@@ -0,0 +1,22 @@
+import * as eks from "@pulumi/eks";
+import * as k8s from "@pulumi/kubernetes";
+import * as pulumi from "@pulumi/pulumi";
+
+export function checkStorageClasses(storageClasses: eks.UserStorageClasses) {
+    storageClasses.apply(scs => {
+        for (const key of Object.keys(scs)) {
+            const sc = scs[key];
+            sc.metadata.apply(m => {
+                if (m.name) {
+                    // TODO: @metral remove this when issue is fixed.
+                    // 'defaultName' is used per suggestion in https://git.io/fjVYi due to
+                    // https://github.com/pulumi/pulumi/issues/2433
+                    const name = m.name || "defaultName";
+                    if (!name || name === "defaultName") {
```

Comment:
The logic here looks very odd. We’re inside an `if m.name`, then we default a value in (which can never happen). Then we check if it’s still undefined which doubly can’t ever happen. 

It feels like this is not accomplishing what you intend, or at least is an awkward way to accomplish it if it is?

Reply:
Done